ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -64,29 +64,17 @@ export default React.createClass({
   getInitialMessages(questionObject){
     var messages = [];
     if(_.has(questionObject, 'encodedText')){
-      var arrayOfMessages = questionObject.encodedText.split('[:');
-      for (var index in arrayOfMessages){
-        var rawMessageText = arrayOfMessages[index];
-        if(rawMessageText !== ''){
-          const messageCode = rawMessageText.substring(0, 2);
-          var messageText = rawMessageText.substring(2, rawMessageText.length);
-          var messageType = undefined;
-          var student = undefined;
-          messageCode === 'i]' ? messageType = 'info' : messageCode === 'u]' ? messageType = 'user' : messageType = 'student';
-          if(messageType === 'student'){
-            const splitStudentMessage = rawMessageText.split('s]');
-            const studentId = Number(splitStudentMessage[0]);
-            messageText = splitStudentMessage[1];
-            student = _.find(this.props.question.students, student => student.id===studentId);
-          }
-          const message = {
-            type: messageType,
-            text: messageText,
-            key: messageText,
-            student
-          };
-          messages.push(message);
-        }
+
+      for (var index in questionObject.encodedText){
+        const textObject = questionObject.encodedText[index];
+        const student = _.find(this.props.question.students, student => student.id === textObject.studentId);
```

Comment:
A few style suggestions for you.

For transforming a list of values, I'd suggest using `.map` over iteration with for loops in general.
```
const messages = question.messages.map((message) => {
  ...
});
```

Also you can use a lodash shorthand for find here that might be more direct:

```
const student = _.find(this.props.question.students, { id: message.studentId});
```